builtInFunctions/esdtDataStorage.go

```diff
@@ -202,32 +207,91 @@ func (e *esdtDataStorage) checkCollectionIsFrozenForAccount(
 	return nil
 }
 
-// SaveESDTNFTToken saves the nft token to the account and system account
-func (e *esdtDataStorage) SaveESDTNFTToken(
-	senderAddress []byte,
+func (e *esdtDataStorage) checkFrozenPauseProperties(
 	acnt vmcommon.UserAccountHandler,
 	esdtTokenKey []byte,
 	nonce uint64,
 	esdtData *esdt.ESDigitalToken,
-	mustUpdate bool,
 	isReturnWithError bool,
-) ([]byte, error) {
+) error {
 	err := checkFrozeAndPause(acnt.AddressBytes(), esdtTokenKey, esdtData, e.globalSettingsHandler, isReturnWithError)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	esdtNFTTokenKey := computeESDTNFTTokenKey(esdtTokenKey, nonce)
 	err = checkFrozeAndPause(acnt.AddressBytes(), esdtNFTTokenKey, esdtData, e.globalSettingsHandler, isReturnWithError)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	err = e.checkCollectionIsFrozenForAccount(acnt, esdtTokenKey, nonce, isReturnWithError)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// AddToLiquiditySystemAcc will increase/decrease the liquidity for ESDT Tokens on the metadata
+func (e *esdtDataStorage) AddToLiquiditySystemAcc(
+	esdtTokenKey []byte,
+	nonce uint64,
+	transferValue *big.Int,
+) error {
+	if !e.flagSaveToSystemAccount.IsSet() || !e.flagSendAlwaysEnableEpoch.IsSet() || nonce == 0 {
+		return nil
+	}
+
+	esdtNFTTokenKey := computeESDTNFTTokenKey(esdtTokenKey, nonce)
+	esdtData, systemAcc, err := e.getESDTDigitalTokenDataFromSystemAccount(esdtNFTTokenKey)
+	if err != nil {
+		return err
+	}
+
+	if esdtData == nil {
+		return ErrNilESDTData
+	}
+
+	// old style metaData - nothing to do
+	if len(esdtData.Reserved) == 0 {
+		return nil
+	}
+
+	esdtData.Value.Add(esdtData.Value, transferValue)
+	if esdtData.Value.Cmp(zero) <= 0 {
```

Comment:
please add a WARN or ERROR log if the count is below 0 as this can indicate a pretty big issue

Reply:
done

Reply:
made a return error - as it would invalidate the transaction